microsoft/mcsa-observability
# Security Wave 1: updated changes

## Purpose

* MSI for Service Bus in TimerStart function (connection string removed)
* Enable MSI ingestion from Storage -> ADX (remove need for SAS token in ingestion)
* Replace blob connection string with MSI
* Remove unused .sh files

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe: Security
```

## How to Test
*  Follow the steps in README
* Open app settings for both AdxIngest and TimerStart Function and edit the msftTenantId property by pasting the Microsoft Tenant Id: 72f988bf-86f1-41af-91ab-2d7cd011db47
* Navigate to the created app registration of this deployment and make sure to add Service Tree info to Branding and Properties and an additional owner to the registration
* Run the following command on the ADX table, replacing the Terraform variable with the Object ID of the ADX system assigned identity
```
.alter-merge cluster policy managed_identity "[{ 'ObjectId' : '${azurerm_kusto_cluster.this.identity.0.principal_id}', 'AllowedUsages' : 'NativeIngestion' }]"
```

## What to Check
Verify that the following are valid
* Ensure data is present in Storage, ADX, and Grafana
